Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -100,28 +100,46 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
     )
     .map { ($0.0, $0.1, $0.2, false) }
 
-    let newCards = self.newStripeIntentCreditCardProperty.signal.skipNil()
-      .map { _, setupIntent in
-        CreatePaymentSourceSetupIntentInput.init(intentClientSecret: setupIntent, reuseable: true)
-      }
-      .switchMap { inputValue in
-        AppEnvironment.current.apiService.addPaymentSheetPaymentSource(input: inputValue)
+    let newlyAddedCardProducer = self.newStripeIntentCreditCardProperty.signal.skipNil()
+      .switchMap { setupIntent, paymentMethodId -> SignalProducer<
+        Signal<UserCreditCards.CreditCard, ErrorEnvelope>.Event,
+        Never
+      > in
+        let input = CreatePaymentSourceSetupIntentInput.init(intentClientSecret: setupIntent, reuseable: true)
+        return AppEnvironment.current.apiService.addPaymentSheetPaymentSource(input: input)
           .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-          .map { (envelope: CreatePaymentSourceEnvelope) in envelope.createPaymentSource }
+          .map { (envelope: CreatePaymentSourceEnvelope) -> UserCreditCards.CreditCard? in
+            guard envelope.createPaymentSource.isSuccessful else {
+              return nil
+            }
+
+            // There is a backend bug where the stripeCardId isn't being refreshed
+            // after the add card mutation is called. This adds it back in.
```

Comment:
nit: It'd be nice to link to a ticket here so we can check if the bug still exists the next time we touch this class, especially if we want to remove this code.